### PR TITLE
Clarify *-pattern rule options in docs

### LINKED ISSUE #1650: Expand regex option documentation across rules
**BUG: selector-class-pattern** 
Does not seem to handle regex expressions properly.

When testing the example as in the documentation: 

```json
"selector-class-pattern": "/foo-[a-z]+/",
````

Both selectors fail, but only the first should fail in theory. 
```css
.foo-BAR {
  color: #FFF;
}

.foo-bar {
  color: #FFF;
}
```

It was wanted to ban following patterns in my css files since the selectors are imported as variables in javascript. 

```css
.test-class {
  color: #FFF;
}
```

And instead allow

```css
.testClass {
  color: #FFF;
}
```

stylelint version: **7.0.1**
stylelint-config-standard version: **11.0.0**

**.stylelintrc:**
```json
{
  "extends": "stylelint-config-standard",
  "rules": {
    "selector-class-pattern": "/[a-z][a-z0-9A-Z]+/",
  }
}
```

The regex `\([a-z][a-z0-9A-Z]+)\` works fine and can be tested at: http://regexr.com/

```css
.testClass {
  color: #00F;
}

.test-class {
	color: #00F;
}
```
**Expected:**
```shell
 x:x  ✖  Expected class selector ".test-class" to match specified pattern   selector-class-pattern
```

**Got:**
```shell
 x:x  ✖  Expected class selector ".testClass" to match specified pattern    selector-class-pattern
 x:x  ✖  Expected class selector ".test-class" to match specified pattern   selector-class-pattern
```

### PR DESCRIPTION
**Pull request by jeddy3**
Closes #1650

I standardised on string options as JSON configs seem most popular.

### CODE REVIEW

**Comment by ntwb** [comment]
I'm thinking rather than `Given:` and then the example we should explicity state that we're using a string in th example, as such maybe `Given the string:`?

**Comment by Nirazul** [comment]
@ntwb 
A few weeks ago we introduced color coding for code blocks on stylelint.io, `Given:` is some sort of keyword that marks a codeblock that is coloured gray. So we'd need to adapt the detection when this should be more flexible.

Here's an [example](http://stylelint.io/user-guide/rules/unit-blacklist/) how it works.

**Comment by ntwb** [comment]
@Nirazul Yes, thanks for that reminder 👍 

An alternate option would be to include a comment in the code example that it is a string rather than changing the `Given:` keyword used in the docs generation if that's more preferable.

**Comment by jeddy3** [comment]
> Given: is some sort of keyword that marks a codeblock that is coloured gray.

Thanks for warning about this, but the "Given:" keyword isn't required when introducing *the first* example within a heading block. As the heading has reset the color classes.

File: src/rules/selector-class-pattern/README.md [hunk 1]
```diff
@@ -68,7 +72,13 @@ div > #zing + .foo-bar {}
 
 This option will resolve nested selectors with `&` interpolation.
 
-### E.g. `/^[A-Z]+$/`
+For example, with `true`.
+
+Given:
```
Comment (ntwb) [inline src/rules/selector-class-pattern/README.md]: One more?

**Comment by ntwb** [comment]
Thanks @jeddy3, looks like one was missed

**Comment by davidtheclark** [comment]
Thanks!

**Pull request merged**